File: consecutive-positive-divisors/consecutive-positive-divisors.js
function consecutivePositiveDivisors(maxNumber){

    let currentNumber = 0, nextNumber = 0, count = 0;

    let newNumberDivisorArray = calculateDivisorsCount(maxNumber);

    for(let i = 2; i < maxNumber; i++){
        currentNumber = i;
        nextNumber = i + 1;

        if(newNumberDivisorArray[currentNumber] === newNumberDivisorArray[nextNumber]){

            count++;
        }
    }

    return count;
}

function calculateDivisorsCount(num){
    let numberDivisorArray = [num+1];

    for(let i = 0; i < num+1; i++){
        numberDivisorArray[i] = 0;
    }

    for(let i = 1; i <= num; i++){
        for(let j = 1; i*j <= num; j++){
            numberDivisorArray[i*j]++;
        }
    }

    return numberDivisorArray;
}

consecutivePositiveDivisors(10_000_000);

File: consecutive-positive-divisors/consecutive-positive-divisors.java
class Main {

   public static int consecutivePositiveDivisors(int maxNumber){
   
       int count = 0;
   
       int[] newNumberDivisorArray = calculateDivisorsCount(maxNumber);
      
       for(int i = 2; i < maxNumber; i++) {
           count = newNumberDivisorArray[i] == newNumberDivisorArray[i+1] ? count + 1 : count;
       }
   
       return count;
   }

   public static int[] calculateDivisorsCount(int num) {
       int[] numberDivisorArray = new int[num+1];
   
       for(int i = 1; i <= num; i++){
           for(int j = 1; i*j <= num; j++){
               numberDivisorArray[i*j]++;
           }
       }
   
       return numberDivisorArray;
   }
   
   public static void main(String[] args) {
      System.out.println(consecutivePositiveDivisors(10000000));
   }

}